forte/data/data_store.py

```diff
@@ -1183,8 +1182,60 @@ def get_length(self, type_name: str) -> int:
             delete_count = self.__deletion_count.get(type_name, 0)
             return len(self.__elements[type_name]) - delete_count
 
+    def _get_bisect_range(
+        self, search_list: SortedList, range_span: Tuple[int]
+    ) -> Optional[List]:
+        """
+        Perform binary search on the specified list for target entry class.
+        Entry class can be a subtype of
+        :class:`~forte.data.ontology.top.Annotation`
+        or :class:`~forte.data.ontology.top.AudioAnnotation`. This function
+        finds the the elements in the `Annotation` or `AudioAnnotation`
+        sorted list whose begin and end index falls within `range_span`.
+
+        Args:
+            search_list: A `SortedList` object on which the binary search
+                will be carried out.
+            range_span: a tuple that indicates the start and end index
+                of the range in which we want to get required entries
+
+        Returns:
+            List of entries to fetch
+        """
+
+        # Check if there are any entries within the given range
+        if (
+            search_list[0][constants.BEGIN_INDEX]
+            > range_span[constants.END_INDEX]
+            or search_list[-1][constants.END_INDEX]
+            < range_span[constants.BEGIN_INDEX]
+        ):
+            return None
+
+        result_list = []
+
+        begin_index = search_list.bisect_left([range_span[0], range_span[0]])
+
+        for idx in range(begin_index, len(search_list)):
+            if (
+                search_list[idx][constants.BEGIN_INDEX]
+                > range_span[constants.END_INDEX]
```

Comment:
Please do not use these `contants` if you don't know what they are for.

Reply:
Sorry I was trying to keep the previous implementation in master (there `range_annotation` was indexed in the same way). I have updated it.